src/es_archiver/lib/indices/kibana_index.js

```diff
@@ -112,26 +130,36 @@ async function loadElasticVersion() {
   return JSON.parse(packageJson).version;
 }
 
-const spacesEnabledCache = new Map();
 export async function isSpacesEnabled({ kibanaUrl }) {
-  if (!spacesEnabledCache.has(kibanaUrl)) {
-    const statuses = await getKibanaStatuses({ kibanaUrl });
-    spacesEnabledCache.set(kibanaUrl, !!statuses.find(({ id }) => id.startsWith('plugin:spaces@')));
-  }
-
-  return spacesEnabledCache.get(kibanaUrl);
+  return await getKibanaPluginEnabled({
+    kibanaUrl,
+    pluginId: 'spaces'
+  });
 }
 
-async function getKibanaStatuses({ kibanaUrl }) {
-  try {
-    const { payload } = await wreck.get('/api/status', {
-      baseUrl: kibanaUrl,
-      json: true
-    });
-    return payload.status.statuses;
-  } catch (error) {
-    throw new Error(`Unable to fetch Kibana status API response from Kibana at ${kibanaUrl}`);
+async function getKibanaPluginEnabled({ pluginId, kibanaUrl }) {
+  if (!kibanaPluginIdsCache.has(kibanaUrl)) {
+    kibanaPluginIdsCache.set(kibanaUrl, (async () => {
+      try {
+        const { payload } = await wreck.get('/api/status', {
+          baseUrl: kibanaUrl,
+          json: true
+        });
+
+        return payload.status.statuses
+          .map(({ id }) => {
+            const match = id.match(/^plugin:(.+)@.*$/, '$1');
+            return match ? match[1] : undefined;
+          })
+          .filter(Boolean);
+
+      } catch (error) {
+        throw new Error(`Unable to fetch Kibana status API response from Kibana at ${kibanaUrl}`);
```

Comment:
Might be nice to log the error, so we can understand why it's failing.

Reply:
Normally the most helpful output is from the Kibana server from above where this error shows up in the test output, but can't hurt to include the message from this error in the output.